File: backend/utils/api_error.py
class APIError(Exception):
    def __init__(
            self,
            status_code: int,
            message: str = "Something went wrong !!...", 
            errors: list = None,
            ):
        super().__init__(message)

        self.status_code = status_code
        self.message = message
        self.success = False
        self.data = None
        self.errors = errors if errors is not None else []
    
    def to_dict(self):
        return {
            "message": self.message,
            "success": self.success,
            "data": self.data,
            "errors": self.errors
        }

File: server/src/utils/apiError.ts
class ApiError extends Error{
    statusCode: number;
    message: string;
    data: null;
    error?:any;
    stack?: string;
    success: boolean;

    constructor(
        statusCode: number,
        message: string,
        error?:any,
        stack?:string
    ){
        super(message)
        this.statusCode = statusCode;
        this.message = message
        this.data = null;
        this.success = false;

        if(error){
            this.error = error
        }
        if(stack){
            this.stack = stack
        }else{
            Error.captureStackTrace(this, this.constructor)
        }
    }
}

export{ ApiError }